astropy/io/fits/file.py

```diff
@@ -145,6 +149,15 @@ def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
         if mode is None:
             mode = 'readonly'
 
+        # Handle remote or cloud URIs with fsspec
+        if use_fsspec:
+            if not HAS_FSSPEC:
+                raise ModuleNotFoundError("please install `fsspec` to access this file")
+            if fsspec_kwargs is None:
+                fsspec_kwargs = {}
+            fileopen = fsspec.open(fileobj, **fsspec_kwargs)
+            fileobj = fileopen.open()
```

Comment:
Does this raise an exception if file is not found? I believe this might fail silently?

Reply:
You should get FileNotFoundError